executor/analyze.go

```diff
@@ -118,7 +119,31 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	// The meaning of key in map is the structure that used to store the tableID and indexID.
 	// The meaning of value in map is some additional information needed to build global-level stats.
 	globalStatsMap := make(map[globalStatsKey]globalStatsInfo)
-
+	finishJobWithLogFn := func(ctx context.Context, job *statistics.AnalyzeJob, meetError bool) {
+		job.Finish(meetError)
+		if job != nil {
+			table, err := infoschema.GetInfoSchema(e.ctx).TableByName(model.NewCIStr(job.DBName), model.NewCIStr(job.TableName))
+			if err != nil {
+				logutil.Logger(ctx).Error("analyze table logging error", zap.String("reason", err.Error()))
+				return
+			}
+			colsName := make([]string, 0, len(table.Cols()))
+			for _, col := range table.Cols() {
+				colsName = append(colsName, col.Name.String())
+			}
+			idxsName := make([]string, 0, len(table.Indices()))
+			for _, idx := range table.Indices() {
+				idxsName = append(idxsName, idx.Meta().Name.String())
+			}
+			logutil.Logger(ctx).Info(fmt.Sprintf("analyze table `%s`.`%s` has %s", job.DBName, job.TableName, job.State),
+				zap.String("partition", job.PartitionName),
+				zap.Strings("columns", colsName),
+				zap.Strings("indices", idxsName),
+				zap.Time("start time", job.StartTime),
+				zap.Time("end time", job.EndTime),
+				zap.String("cost", job.EndTime.Sub(job.StartTime).String()))
+		}
```

Comment:
I just checked the related content, I found that we don’t actually need to go to `information_schema` to get detailed information, `job.DBName`, `job.TableName` and `job.JobInfo` are enough.

Now writing this way will cause the statement that is actually `analyze index` to print irrelevant column information.

I'm sorry that this information was not provided before you started.

Reply:
appreciate for your reminder 😆